### PR TITLE
243 aggregator signature aggregator mixes batches making the agg response fail

### LINKED ISSUE #243: ( Aggregator ): Signature aggregator mixes batches, making the agg response fail

### PR DESCRIPTION
**Pull request by MauroToscano**
Fixed the bug by having maps with Roots and Indexes, and not assuming the  Signature is always working over the last element. 

Additionally I removed some data that wasn't being used or signed from the SignedResponse of the operator, changed the contract pragma so it can compile, and renamed a variable from task to batch for consistency

### CODE REVIEW

File: aggregator/internal/pkg/aggregator.go [hunk 1]
```diff
@@ -193,44 +210,65 @@ func (agg *Aggregator) sendAggregatedResponseToContract(blsAggServiceResp blsagg
...
+	batchMerkleRoot := agg.batchesRootByIdx[blsAggServiceResp.TaskIndex]
+	agg.batchesRootByIdxMutex.Unlock()
 
 	_, err := agg.avsWriter.SendAggregatedResponse(context.Background(), batchMerkleRoot, nonSignerStakesAndSignature)
 	if err != nil {
 		agg.logger.Error("Aggregator failed to respond to task", "err", err)
 	}
 }
 
+// BatchIndex
```
Comment (entropidelic) [inline aggregator/internal/pkg/aggregator.go]: small nit, but I think this is not needed

File: contracts/src/core/ERC20Mock.sol [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: MIT
 // OpenZeppelin Contracts (last updated v4.8.0) (token/ERC20/ERC20.sol)
 
-pragma solidity ^0.8.9;
+pragma solidity =0.8.12;
```
Comment (entropidelic) [inline contracts/src/core/ERC20Mock.sol:4]: is this change ok? seems unrelated to the PR

File: aggregator/internal/pkg/aggregator.go (hunk 1 above)
Comment (MauroToscano) [reply aggregator/internal/pkg/aggregator.go]: I missed it, thanks

**Review by entropidelic — APPROVED** [review]
lgtm, just left some small nits

File: contracts/src/core/ERC20Mock.sol (hunk 2 above)
Comment (MauroToscano) [reply contracts/src/core/ERC20Mock.sol:4]: It is, but make bindings is not working on main, I solved some small issues

Comment (entropidelic) [reply contracts/src/core/ERC20Mock.sol:4]: great, thanks

**Review by NicolasRampoldi — APPROVED** [review]
lgtm, tested it and it works as expected

**Pull request merged**